codeguru_profiler_agent/agent_metadata/agent_debug_info.py

```diff
@@ -0,0 +1,124 @@
+        }
+
+    @synchronized
+    def increment_sdk_error(self, error_type):
+        """
+        ErrorsCount is the umbrella of all the kinds of error we want to capture. Currently we have only SdkClientErrors
+        in it. SdkClientErrors is comprised of different API level errors like ConfigureAgentErrors,
+        PostAgentProfileErrors, CreateProfilingGroupErrors.
+        :param error_type: The type of API level error that we want to capture.
+        """
+        self.errors_count += 1
+        self.sdk_client_errors += 1
+
+        """
+        Special handling for ResourceNotFoundException errors.
+        For example configureAgentRnfeAutoCreateEnabledErrors is also a configureAgentErrors.
+        """
+        if error_type == "configureAgentErrors":
+            self.configure_agent_errors += 1
+        elif error_type == "configureAgentRnfeAutoCreateEnabledErrors":
+            self.configure_agent_errors += 1
+            self.configure_agent_rnfe_auto_create_enabled_errors += 1
+        elif error_type == "createProfilingGroupErrors":
+            self.create_profiling_group_errors += 1
+        elif error_type == "postAgentProfileErrors":
+            self.post_agent_profile_errors += 1
+        elif error_type == "postAgentProfileRnfeAutoCreateEnabledErrors":
+            self.post_agent_profile_errors += 1
+            self.post_agent_profile_rnfe_auto_create_enabled_errors += 1
+
+    def record_sdk_error(self, error_type):
+        self.increment_sdk_error(error_type)
+
+
+class AgentDebugInfo:
+    def __init__(self, errors_metadata=None, agent_start_time=None, timer=None):
+        self.process_id = get_process_id()
+        self.errors_metadata = errors_metadata
+        self.agent_start_time = agent_start_time
+        self.timer = timer
+
+    def serialize_to_json(self):
+        """
+        This needs to be compliant with agent debug info schema.
+        """
+        json = {}
+
+        self.add_agent_start_time(json)
+        self.add_process_id(json)
+        self.add_errors_metadata(json)
+        self.add_generic_metrics(json)
+
+        return json
+
+    def add_agent_start_time(self, json):
+        if self.agent_start_time is not None:
+            json["agentStartTime"] = int(self.agent_start_time)
+
+    def add_errors_metadata(self, json):
+        if self.errors_metadata is not None:
+            json["errorsCount"] = self.errors_metadata.serialize_to_json()
+
+    def add_process_id(self, json):
+        if self.process_id is not None:
+            json["processId"] = self.process_id
+
+    def add_generic_metrics(self, json):
+        if self.timer is not None and self.timer.metrics:
+            generic_metrics = {}
+
+            for metric in self.timer.metrics:
+                metric_value = self.timer.metrics[metric]
+                generic_metrics[metric + "_max"] = metric_value.max
```

Comment:
should you not add "_timings" to be consistent with the java agent?

Reply:
Yes. Thanks for pointing it out. Fixed it.